h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -539,6 +618,34 @@ protected double updateState(double [] beta,GLMGradientInfo ginfo){
     return (_relImprovement = (objOld - objective())/Math.abs(objOld));
   }
 
+  protected void setHGLMComputationState(double [] beta, double[] ubeta, double[] psi, double[] phi, 
+                                         double hlcorrection, double tau, Frame wpsi, String[] randCoeffNames){
+    _beta = copyArray(beta, _beta);
+    _ubeta = copyArray(ubeta, _ubeta);
+    _randCoeffNames = copyArray(randCoeffNames, _randCoeffNames);
+    _psi = copyArray(psi, _psi);
+    _phi = copyArray(phi, _phi);
+    _correction_HL = hlcorrection;
+    _tau = tau;
+    _priorw_wpsi = wpsi;  // store prior_weight and calculated wpsi value for coefficients of random columns
+    _iterHGLM_GLMMME = 0;
+  }
+  
+  private double[] copyArray(double[] source, double[] dest) {
```

Comment:
For these two methods, it looks like we could use built-in `Arrays.copyOf`. If not and you really need a method to copy into an existing instance,  may I suggest moving these methods to one of `ArrayUtils` classes we have in H2O.

As a matter of personal preference, if I'd have been allocating primitive array types, I'd use `MemoryManager` - the more H2O memory manager knows about, the better I think.

Reply:
Agree on both points.  Done.